featuretools/primitives/standard/transform_primitive.py

```diff
@@ -906,3 +907,138 @@ def lag(time_index, numeric):
             x = pd.Series(numeric.values, index=time_index.values)
             return x.shift(periods=self.periods, fill_value=self.fill_value).values
         return lag
+
+
+class IsInGeoBox(TransformPrimitive):
+    """Determines if coordinates are inside a box defined by two
+    corner coordinate points.
+
+    Description:
+        Coordinate values should be specified as (latitude, longitude)
+        tuples. This primitive is unable to handle coordinates and boxes
+        at the poles, and near +/- 180 degrees latitude.
+
+    Args:
+        point1 (tuple(float, float)): The coordinates
+            of the first corner of the box. Defaults to (0, 0).
+        point2 (tuple(float, float)): The coordinates
+            of the diagonal corner of the box. Defaults to (0, 0).
+
+    Example:
+        >>> is_in_geobox = IsInGeoBox((40.7128, -74.0060), (42.2436, -71.1677))
+        >>> is_in_geobox([(41.034, -72.254), (39.125, -87.345)]).tolist()
+        [True, False]
+    """
+    name = "is_in_geobox"
+    input_types = [ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+
+    def __init__(self, point1=(0, 0), point2=(0, 0)):
+        self.point1 = point1
+        self.point2 = point2
+        self.lats = np.sort(np.array([point1[0], point2[0]]))
+        self.lons = np.sort(np.array([point1[1], point2[1]]))
+
+    def get_function(self):
+        def geobox(latlongs):
+            if latlongs.hasnans:
+                latlongs = np.where(latlongs.isnull(), pd.Series([(np.nan, np.nan)] * len(latlongs)), latlongs)
+            transposed = np.transpose([list(latlon) for latlon in latlongs])
+            lats = (self.lats[0] <= transposed[0]) & \
+                   (self.lats[1] >= transposed[0])
```

Comment:
bit of a readability nitpick - might be nice to do something like `southern_lat, northern_lat = self.lats` (and eastern/western for the longitudes--though if that's confusing smaller/larger would also work) and use those variables, which may make it clearer in building the `lats` and `longs` parameters that we're filtering for points that are between the specified points

Reply:
I tried this, and it actually makes ti more confusing because you have 4 variables, instead of 2. So going to keep this as is.

Reply:
Sounds good! Thanks for hearing me out